Autopilot/AttitudeManager/AttitudeManager.c

```diff
@@ -693,137 +644,148 @@ void readDatalink(void){
                 break;
             case SET_IMU:
                 setVNOrientationMatrix(CMD_TO_FLOAT_ARRAY(cmd->data));
+                break;    
+            case SET_ROLL_RATE_GAINS:
+                setGain(ROLL_RATE, KP, CMD_TO_FLOAT(cmd->data));
+                setGain(ROLL_RATE, KD, CMD_TO_FLOAT(cmd->data + 4));
+                setGain(ROLL_RATE, KI, CMD_TO_FLOAT(cmd->data + 8));
                 break;
-            case SET_KDVALUES:
-                setKValues(KD, CMD_TO_FLOAT_ARRAY(cmd->data));
+            case SET_PITCH_RATE_GAINS:
+                setGain(PITCH_RATE, KP, CMD_TO_FLOAT(cmd->data));
+                setGain(PITCH_RATE, KD, CMD_TO_FLOAT(cmd->data + 4));
+                setGain(PITCH_RATE, KI, CMD_TO_FLOAT(cmd->data + 8));
                 break;
-            case SET_KPVALUES:
-                setKValues(KP, CMD_TO_FLOAT_ARRAY(cmd->data));
+            case SET_YAW_RATE_GAINS:
+                setGain(YAW_RATE, KP, CMD_TO_FLOAT(cmd->data));
+                setGain(YAW_RATE, KD, CMD_TO_FLOAT(cmd->data + 4));
+                setGain(YAW_RATE, KI, CMD_TO_FLOAT(cmd->data + 8));
                 break;
-            case SET_KIVALUES:
-                setKValues(KI, CMD_TO_FLOAT_ARRAY(cmd->data));
+            case SET_ROLL_ANGLE_GAINS:
+                setGain(ROLL_ANGLE, KP, CMD_TO_FLOAT(cmd->data));
+                setGain(ROLL_ANGLE, KD, CMD_TO_FLOAT(cmd->data + 4));
+                setGain(ROLL_ANGLE, KI, CMD_TO_FLOAT(cmd->data + 8));
                 break;
-            case SET_GAINS:
-            {
-                char channel = cmd->data[0];
-                //We're making the rest of the data word aligned here, which is required for casting it to floats
-                memcpy(cmd->data, cmd->data + 1, cmd->data_length - 1);
-                setGains(channel, CMD_TO_FLOAT_ARRAY(cmd->data));
```

Comment:
That's not arbitrary gains for all channels. What if I wanted to set Ki for altitude control? or add a little bit of Kd to the ground speed controller?